test/sql/test_deprecations.py

```diff
@@ -135,6 +156,104 @@ def test_table_useexisting(self):
                 autoload_with=testing.db,
             )
 
+    def test_case_sensitive(self):
+        reg = functions._registry['_default']
+        cs_reg = functions._case_sensitive_registry['_default']
+
+        class MYFUNC(GenericFunction):
+            type = DateTime
+
+        assert isinstance(func.MYFUNC().type, DateTime)
+        assert isinstance(func.MyFunc().type, DateTime)
+        assert isinstance(func.mYfUnC().type, DateTime)
+        assert isinstance(func.myfunc().type, DateTime)
+
+        in_("myfunc", reg)
+        not_in_("MYFUNC", reg)
+        not_in_("MyFunc", reg)
+        in_("myfunc", cs_reg)
+        eq_(set(cs_reg['myfunc'].keys()), set(['MYFUNC']))
+
+        with testing.expect_deprecated():
```

Comment:
**mike bayer** (zzzeek) wrote:

this is note [1], this isn't actually asserting that a deprecation is being emitted unless at least the beginning of the deprecation message is put here; there's no warning actually being triggered right now here.

Reply:
Ups, I didn't see that. I updated the `expect_*` statements.

Reply:
Ups, I didn't see that. I updated the `expect_*` statements.